src/ddmd/expression.d

```diff
@@ -11007,10 +11007,7 @@ extern (C++) final class NegExp : UnaExp
         if (tb.ty == Tarray || tb.ty == Tsarray)
         {
             if (!isArrayOpValid(e1))
-            {
-                error("invalid array operation %s (possible missing [])", toChars());
-                return new ErrorExp();
-            }
+                return arrayOpInvalidError(this);
```

Comment:
The refactoring seems fine, but it should have been done in a separate commit or PR.